src/index.test.tsx

```diff
@@ -15,13 +27,132 @@ describe('React Hook Form Input', () => {
         register={register}
         unregister={unregister}
         name="test"
-        as={<input />}
+        as="input"
+      />,
+    );
+
+    expect(asFragment).toMatchSnapshot();
+  });
+
+  it('should render MUI Checkbox correctly', () => {
+    const setValue = () => {};
+    const register = () => () => {};
+    const unregister = () => {};
+    const { asFragment } = render(
+      <RHFInput
+        as={Checkbox}
+        name="Checkbox"
+        type="checkbox"
+        value="test"
+        unregister={unregister}
+        register={register}
+        setValue={setValue}
       />,
     );
 
     expect(asFragment).toMatchSnapshot();
```

Comment:
that was my bad LOL. `asFragment()` and you inherited them

Reply:
I just mimicked on how you render component correctly.